Assets/Scripts/Utility/AssetInjection/Components/CustomDoor.cs

```diff
@@ -26,10 +26,13 @@ public class CustomDoor : MonoBehaviour
     {
         private StaticDoor? staticDoor;
 
+        [Tooltip("Type of door that is being defined. None means building textures are used to determine this.")]
+        public DoorTypes DoorType = DoorTypes.None;
```

Comment:
There are only three types of door: `Building`, `DungeonEntrance` and `DungeonExit`. Choosing anything other than the correct one results in a broken door, in one case even with an exception being thrown. Can you explain what's the intended use case here?

Reply:
Please take a look at line 95 to see how I use this variable. I check if this variable is set to anything other than `None` and if so, the trigger door's type is set to whatever value the modder set. If it is set to `None`, the DoorType doesn't get touched and it will remain what it was set to by the aforementioned GetModelData function. There is no possibility for my changes to set the door type to `None`. If you're worried about a modder choosing the wrong DoorType for their model of the three valid ones, I think it's their responsibility to either make sure it's correct or just keep it set to `None` and let the game decide what type they want.

As for the intended use case: In general, I want the ability to override the GetModelData function's decision about what type of door I'm placing. A specific example is the scenario I said in my other comment. Another one I can think of is if the modder is adding doors to a model that didn't originally have doors on it. Unless the game already has some failsafes in place, I think the DoorType of the door written into staticDoor will be `None` unless the modder uses my override.